src/libs/signer.service.ts

```diff
@@ -1,17 +1,24 @@
-import { ethers } from "ethers";
+import { BigNumber, ethers } from "ethers";
 
 import { getProvider } from "./ethers.service";
-import { SIGNER_PRIVATE_KEY } from "../globals";
+import { logger } from "./logger.service";
+import { CHAIN_ID, SIGNER_PRIVATE_KEY } from "../globals";
 
-export interface SigningRequest {
+interface SigningRequest {
   transactionData: string;
   to: string;
-  gasLimit: string | number;
+  gasLimit: BigNumber;
```

Comment:
I think number is fine here, it just moves where the big number is converted to a number. BigNumber not a native type so its keeps it more generic